pkgs/build-support/agda/default.nix

```diff
@@ -2,14 +2,28 @@
 
 { stdenv, lib, self, Agda, runCommand, makeWrapper, writeText, ghcWithPackages, nixosTests }:
 
-with lib.strings;
-
 let
+  inherit (lib)
+    attrValues
+    elem
+    filter
+    filterAttrs
+    isAttrs
+    isList
+    platforms
+    ;
+
+  inherit (lib.strings)
+    concatMapStrings
+    concatMapStringsSep
+    optionalString
+    ;
+
   withPackages' = {
     pkgs,
-    ghc ? ghcWithPackages (p: with p; [ ieee754 ])
+    ghc ? ghcWithPackages (p: [ p.ieee754 ])
```

Comment:
I don't think that was wrong as it was. Just as with `rec`, `with` (or something *like* with) is not per se bad and *can* improve clarity under certain conditions - specifically when the surprising scoping behavior is actively avoided.

Reply:
Moved. Thanks.